packages/reqresp/src/ReqResp.ts

```diff
@@ -92,20 +92,21 @@ export class ReqResp {
    */
   async registerProtocol(protocol: Protocol, opts?: ReqRespRegisterOpts): Promise<void> {
     const protocolID = this.formatProtocolID(protocol);
+    const {handler: _handler, inboundRateLimits, ...rest} = protocol;
+
+    if (inboundRateLimits) {
+      // Rate limits can change across hard forks and must always be updated
+      this.rateLimiter.setRateLimits(protocolID, inboundRateLimits);
+    }
```

Comment:
When fork boundary is changed, Now the `getProtocolsAtFork` will not return the new `protocol` object for each protocol id as it's fork aware now. Hence will it not be initialized with the new quota already?

Reply:
Yes, that's how it worked before already but we would not register it again
https://github.com/ChainSafe/lodestar/blob/ee9d4fa72b9b753541e05ca391492ed4a239cdfb/packages/reqresp/src/ReqResp.ts#L97-L98

The protocol ID does not include fork information
https://github.com/ChainSafe/lodestar/blob/ee9d4fa72b9b753541e05ca391492ed4a239cdfb/packages/reqresp/src/utils/protocolId.ts#L6-L7

This is the first time the protocol changed without a version bump which I think would have been the better way to handle this as it would just be a newly registered protocol